src/peer_digest.cc

```diff
@@ -445,16 +436,15 @@ peerDigestHandleReply(void *data, StoreIOBuffer receivedData)
 
 /// handle HTTP response headers in the initial storeClientCopy() response
 static int
-peerDigestFetchReply(void *data, char *buf, ssize_t size)
+peerDigestFetchReply(void *data, ssize_t size)
 {
     DigestFetchState *fetch = (DigestFetchState *)data;
-    PeerDigest *pd = fetch->pd;
-    assert(pd && buf);
+    const auto pd = fetch->pd.get();
```

Comment:
I think we should move this initialization lower, right after peerDigestFetchedEnough(), so that we can assert that `pd` is not nil before dereferencing it.

Reply:
Not done? And if we only dereference `pd` once, in debugs() call, then I would remove `pd` variable completely.